### PR TITLE
[JBPM-10015] Allow to set suspendUntil during user task suspend operation as parameter.

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-10015

depends on https://github.com/kiegroup/jbpm/pull/2100

### CODE REVIEW

**Review by gmunozfe — COMMENTED** [review]
Looks good to me, some minor comments and one typo in the Api params for swagger
well done @elguardian !

**Comment by gmunozfe** [comment]
jenkins retest this

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/UserTaskResource.java [hunk 1]
```diff
@@ -517,11 +517,15 @@ public Response stop(@Context HttpHeaders headers,
     public Response suspend(@Context HttpHeaders headers, 
             @ApiParam(value = "container id that task instance belongs to", required = true, example = "evaluation_1.0.0-SNAPSHOT") @PathParam(CONTAINER_ID) String containerId,
             @ApiParam(value = "identifier of the task instance that should be suspended", required = true, example = "123") @PathParam(TASK_INSTANCE_ID) Long taskId, 
-            @ApiParam(value = "optional user id to be used instead of authenticated user - only when bypass authenticated user is enabled", required = false) @QueryParam("user") String userId) {
+            @ApiParam(value = "optional user id to be used instead of authenticated user - only when bypass authenticated user is enabled", required = false) @QueryParam("user") String userId,
+            @ApiParam(value = "priority as Integer", required = false, examples=@Example(value= {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/UserTaskResource.java:521]: done

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/UserTaskServicesClientImpl.java [hunk 2]
```diff
@@ -325,11 +325,16 @@ public void stopTask(String containerId, Long taskId, String userId) {
...
 
-            sendTaskOperation(containerId, taskId, TASK_URI + "/" + TASK_INSTANCE_SUSPEND_PUT_URI, getUserQueryStr(userId));
+    @Override
+    public void suspendTask(String containerId, Long taskId, String userId, Map<String, Object> parameters) {
+        if( config.isRest() ) {
+            sendTaskOperation(containerId, taskId, TASK_URI + "/" + TASK_INSTANCE_SUSPEND_PUT_URI, getUserQueryStr(userId), parameters);
         } else {
-            CommandScript script = new CommandScript( Collections.singletonList( (KieServerCommand) new DescriptorCommand( "UserTaskService", "suspend", new Object[]{containerId, taskId, userId}) ) );
+            CommandScript script = new CommandScript( Collections.singletonList( (KieServerCommand) new DescriptorCommand( "UserTaskService", "suspend", 
+                    serialize(parameters), marshaller.getFormat().getType(),  new Object[]{containerId, taskId, userId}) ) );
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/UserTaskServicesClientImpl.java:337]: done

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java [hunk 3]
```diff
@@ -370,6 +370,34 @@ public void testSuspendAndResume() throws Exception {
...
+            List<TaskSummary> taskList = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+            assertNotNull(taskList);
+
+            assertEquals(1, taskList.size());
+            TaskSummary taskSummary = taskList.get(0);
+            checkTaskNameAndStatus(taskSummary, "First task", Status.Reserved);
+
+            // release task
+            taskClient.suspendTask(CONTAINER_ID, taskSummary.getId(), USER_YODA, Collections.singletonMap("suspendUntil", "PT3S"));
+            Thread.sleep(6000L);
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java:388]: I can do polling.... but thread sleep will be there anyway.

**Comment by elguardian** [comment]
Jenkins retest this please

**Pull request merged**